docs/architecture.md

```diff
@@ -18,3 +18,94 @@
   subscribe to the Redux store without React, but this would be more complicated.
 
 ## Back-end architecture
+
+![Diagram](images/architecture-back-end.drawio.svg)
+
+### Dataset files
+
+All persistent data is stored on the back-end server as JSON files, in the following folder structure
+as seen from the SERVER_DATA_ROOT location:
+
+```
+├── datasets
+│   ├── some-dataset
+│   │   ├── locations.json (array of lng-lat coordinates for each initiative)
+│   │   ├── searchable.json (array of the property values and searchable strings for each initiative)
+│   │   ├── initiatives
+│   │   |   ├── 0.json (full info of first initiative in the above aggregate JSONs)
+│   │   |   ├── 1.json
+│   │   |   ├── ...
+│   ├── other-dataset
+│   │   ├── ...
+│   ├── ...
+```
+
+Additionally, for each map variant, there's a `config.json`, which contains config for displaying the map in the UI, including the vocabs (translations of data IDs), default sidebar panel, and popup appearance.
+
+### Example file contents
+
+`locations.json`:
+
+```
+[ [1.21419, 50.45254], [0.21002, 49.33954], … ]
+```
+
+`searchable.json`:
+
+```
+[ { "coun": "coun:GB", "sz": "sz:small", "searchString": "some co-op 2 green lane london n4 9qr" }, … ]
+```
+
+`0.json`:
+
+```
+{ name: "Some Co-op", "desc": "A co-op that sells stuff", "lng": 1.21419, "lat": 50.45254, "coun": "coun:GB", "sz": "sz:small", ... }
+```
+
+`config.json`:
+
+```
+{
+  "prefixes": {
+    "https://example.com/sizes/1.1/": "sz",
+    ...
+  },
+  "vocabs": {
+    "sz:": {
+      "EN": {
+        "title": "Sizes",
+        "terms": {
+          "sz:large": "Large",
+          "sz:medium": "Medium",
+          "sz:small": "Small"
+        }
+      }
+    },
+    ...
+  },
+  "popups": { "field": "sz", "style": "text" }, { "field": "websites", "style": "clickable-list" },
+  "ui": { ... },
+  ...
+}
+
+```
+
+### Data generation
+
+These directories of JSONs, including the searchable strings in the `searchable.json` files, need to be pre-generated by a script. This script will be written in JS/TS and live in the monorepo, to be run on the back-end server.
+
+The script will take the full data CSV for a map (generated by the data factory) as inputs, and write the full data into the required JSON files in the directory structure specified above.
+
+#### Note:
+
+We will need to manually copy the `standard.csv` from the data factory server to the back-end. Maybe in the future, the data factory pipeline can be enhanced to write the JSON files to the back-end server so that no manual duplication is necessary (and maybe we can eventually get rid of the separate data server altogether).
```

Comment:
Or, perhaps better, the mykomap server could be given an URL to the appropriate standard.csv file(s) as published by the data factory and download it from there (possibly when notified by a webhook, or possibly polling and checking the file modification date)

Reply:
Added to the docs